File: stepik-java-api/src/main/java/org/stepik/api/objects/steps/Sample.java
package org.stepik.api.objects.steps;

/**
 * @author meanmail
 */
public class Sample {
    private String input;
    private String output;

    public String getInput() {
        return input;
    }

    public void setInput(String input) {
        this.input = input;
    }

    public String getOutput() {
        return output;
    }

    public void setOutput(String output) {
        this.output = output;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;

        Sample strings = (Sample) o;

        //noinspection SimplifiableIfStatement
        if (input != null ? !input.equals(strings.input) : strings.input != null) return false;
        return output != null ? output.equals(strings.output) : strings.output == null;
    }

    @Override
    public int hashCode() {
        int result = super.hashCode();
        result = 31 * result + (input != null ? input.hashCode() : 0);
        result = 31 * result + (output != null ? output.hashCode() : 0);
        return result;
    }
}

File: stepik-java-api/src/main/java/org/stepik/api/objects/steps/Sample.kt
package org.stepik.api.objects.steps

import java.util.*


class Sample : ArrayList<String>() {

    var input: String
        get() {
            if (isEmpty()) {
                return ""
            }
            return first()
        }
        set(value) {
            if (isEmpty()) {
                add(value)
            } else {
                set(0, value)
            }
        }

    var output: String
        get() {
            if (size < 2) {
                return ""
            }
            return get(1)
        }
        set(value) {
            when {
                isEmpty() -> {
                    add("")
                }
                size == 1 -> add(value)
                else -> set(1, value)
            }
        }
}
